src/jvmMain/kotlin/io/ktor/io/DirectByteBufferPool.kt

```diff
@@ -21,4 +20,14 @@ public class DirectByteBufferPool(
         check(instance.capacity() == bufferSize)
         check(instance.isDirect)
     }
+
+    public companion object {
+        public val Default: ObjectPool<ByteBuffer> get() = Direct
+
+        public val Direct: ObjectPool<ByteBuffer> = ByteBufferPool(direct = true)
+
+        public val Heap: ObjectPool<ByteBuffer> = ByteBufferPool(direct = false)
+
+        public val NoPool: ObjectPool<ByteBuffer> = ByteBufferPool(capacity = 0)
```

Comment:
differs from other `NoPool` impls, as it will return newly created instance every time, instead of throwing an error

Reply:
The implenentation is right. Should we select a different name?

Reply:
Yes, I think so
Looks like here it should be `EmptyPool` (or just `Empty` if it's in pool companion object) - so no pooling, every call to borrow will create new instance of specified pool buffer size
and `NoPool` (or even some other naming) - pool, when on calling borrow you will get an exception - as now such pools existed for byte arrays. So it's like a dummy pool which you need to use, when you create buffer from you own unpoolable data, so you will be able to recycle buffer to nowhere, and you will not be able to borrow from it